src/builtin.c

```diff
@@ -1879,6 +1879,32 @@ BINOPS
 #undef LIBM_DD
 #undef LIBM_DA
 
+// This is a hack to make last(empty) yield no output values without using boxing.
+static block gen_last_1() {
+  block last_var = gen_op_var_fresh(STOREV, "last");
+  block is_empty_var = gen_op_var_fresh(STOREV, "is_empty");
+  block init = BLOCK(gen_op_simple(DUP),
+                     gen_const(jv_null()),
+                     last_var,
+                     gen_op_simple(DUP),
+                     gen_const(jv_true()),
+                     is_empty_var);
+  block call_arg = BLOCK(gen_call("arg", gen_noop()),
+                         gen_op_simple(DUP),
+                         gen_op_bound(STOREV, last_var),
+                         gen_const(jv_false()),
+                         gen_op_bound(STOREV, is_empty_var),
+                         gen_op_simple(BACKTRACK));
+  block if_empty = gen_op_simple(BACKTRACK);
+  return BLOCK(init,
+               gen_op_target(FORK, call_arg),
+               call_arg,
+               BLOCK(gen_op_bound(LOADVN, is_empty_var),
+                     gen_op_target(JUMP_F, if_empty),
+                     if_empty,
+                     gen_op_bound(LOADVN, last_var)));
```

Comment:
Trying to understand what kind of "pesudo"-jq this translates, will
```
last(g)
```
be something like (vm is not limited by scoping rules?)
```
true as $is_empty | (g as $last | false as $is_empty | empty) // if $is_empty then empty else $last end
```

Reply:
Closer to write with `reduce`;
```jq
def last(g): reduce g as $item (
  { last: null, is_empty: true };
  .last = $item | .is_empty = false
) | if .is_empty then empty else .last end;
```